### PR TITLE
clickhouse_user: set default role

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
See the changes

### CODE REVIEW

**Comment by Andersson007** [comment]
cc @aleksvagachev

**Comment by Andersson007** [comment]
wait a minute, got some errors locally, fixing

**Comment by Andersson007** [comment]
I'll finish it after PTO

**Comment by Andersson007** [comment]
cc @aleksvagachev it's ready for review

File: plugins/modules/clickhouse_user.py [hunk 1]
```diff
@@ -94,6 +131,41 @@
...
+    default_roles: []
+
+- name: Revoke all roles from alice
+  community.clickhouse.clickhouse_user:
+    login_host: localhost
+    login_user: alice
+    login_db: foo
+    login_password: my_password
+    name: test_user
+    roles: []
```
[the remark this answers is not in the document]
Comment (Andersson007) [reply plugins/modules/clickhouse_user.py]: @aleksvagachev good catch, fixed, thanks, PTAL

**Comment by Andersson007** [comment]
i'm now thinking that maybe that `append_role` thing that used say in `ansible.builtin.user` module can be made more flexible? how about something like:
```
role_mode: append | remove | listed_only (default)
default_role_mode:  append | remove | listed_only (default)
```
On one hand that append_role it's how some of long-existing modules handle it - not saying it's good per se and probably it's not the best design but people are used to it. On the other hand, something like this feels more flexible.
What do you think @aleksvagachev ? Maybe you have a third option on your mind?

**Comment by aleksvagachev** [comment]
@Andersson007 I think it's interesting to add this. There is an add functionality, but there is no delete functionality)
It is probably necessary to form the idea that the module can perform only basic actions. And for more specific settings, it is better to use the clickhouse_client module. Otherwise, sooner or later it will become difficult to maintain the code, and users will get confused.

**Comment by Andersson007** [comment]
> And for more specific settings, it is better to use the clickhouse_client module. Otherwise, sooner or later it will become difficult to maintain the code, and users will get confused.

I think the latest suggested implementation is OK in this particular case that it'll basically add only `remove`, so will not complicate the code much. Let me try

**Comment by Andersson007** [comment]
@aleksvagachev changed, the tests were green, please take another look

**Review by aleksvagachev — APPROVED** [review]

**Comment by aleksvagachev** [comment]
@Andersson007 Cool feature! Thanks for the contribution.

**Comment by Andersson007** [comment]
@aleksvagachev the community is always welcome:) thanks for reviewing and merging!

**Pull request merged**